src/util/piped_process.h

```diff
@@ -80,7 +81,10 @@ class piped_processt
   /// Initiate a new subprocess with pipes supporting communication
   /// between the parent (this process) and the child.
   /// \param commandvec The command and arguments to create the process
-  explicit piped_processt(const std::vector<std::string> &commandvec);
+  /// \param message_handler Optional message handler for logging debug messages
+  explicit piped_processt(
+    const std::vector<std::string> &commandvec,
+    message_handlert *message_handler = nullptr);
 
```

Comment:
We don't have that many users of `piped_processt` just yet, so I'd suggest you just make the `message_handlert` mandatory for now and have the call site use a `null_message_handlert` if they prefer to discard output. I'd rather not see pointers being used here, and the `make_log` dance is a lot of code for very little gain.

Reply:
There are a number of usages in the unit tests. But I get your point. I will remove the `make_log` jig.

Reply:
In the unit tests you might even make use of `null_message_handler` from `unit/testing-utils/message.h`. That way you wouldn't even need to introduce an extra local object, reducing the code churn a bit.

Reply:
Already ahead of you there.